test/middleware/IllnessFeatureMiddleware.test.ts

```diff
@@ -0,0 +1,86 @@
+import 'reflect-metadata';
+
+import Substitute, { Arg } from '@fluffy-spoon/substitute';
+import { expect } from 'chai';
+import { NextFunction, Request, Response } from 'express';
+import { MOVED_TEMPORARILY, OK } from 'http-status-codes';
+import { after, before } from 'mocha';
+import request from 'supertest';
+
+import { IllnessReasonFeatureMiddleware } from 'app/middleware/IllnessReasonFeatureMiddleware';
+import { ENTRY_PAGE_URI, ILLNESS_START_DATE_PAGE_URI } from 'app/utils/Paths';
+
+import { createApp } from 'test/ApplicationFactory';
+
+let initialIllnessReasonFeatureFlag: string | undefined;
+
+describe('Illness Reason Feature Toggle Middleware', () => {
+
+    before(() => {
+        initialIllnessReasonFeatureFlag = process.env.ILLNESS_REASON_FEATURE;
+    });
+
+    after(() => {
+        process.env.ILLNESS_REASON_FEATURE = initialIllnessReasonFeatureFlag;
+    });
+
+    describe('Feature switched on', () => {
+
+        it('should call next() if feature flag is on', () => {
+
+            process.env.ILLNESS_REASON_FEATURE = '1';
```

Comment:
does `@before` get triggered before the test starts? if you set the process.env here, whats the points of the:


    before(() => {
        initialIllnessReasonFeatureFlag = process.env.ILLNESS_REASON_FEATURE;
    });

    after(() => {
        process.env.ILLNESS_REASON_FEATURE = initialIllnessReasonFeatureFlag;
    });

Reply:
The before() and after() tests retain the env property value. 
Each time we set `process.env.ILLNESS_REASON_FEATURE=0` we override this value for the application, so having the after() test allows us to reset the value to what it was before